system/jlib/jptree.cpp

```diff
@@ -3038,51 +3037,56 @@ void PTree::serialize(MemoryBuffer &tgt)
 
 void PTree::deserialize(MemoryBuffer &src)
 {
-    deserializeSelf(src);
+    DeserializeContext deserializeContext;
+    deserializeSelf(src, deserializeContext);
 
-    StringAttr eName;
     for (;;)
     {
         size32_t pos = src.getPos();
-        src.read(eName);
-        if (eName.isEmpty())
+        deserializeContext.name.clear();
+        src.read(deserializeContext.name);
+        if (deserializeContext.name.isEmpty())
             break;
         src.reset(pos); // reset to re-read tree name
         IPropertyTree *child = create(src);
-        addPropTree(eName, child);
+        addPropTree(deserializeContext.name, child);
     }
 }
 
-void PTree::deserializeSelf(MemoryBuffer &src)
+void PTree::deserializeSelf(MemoryBuffer &src, DeserializeContext &deserializeContext)
 {
-    setName(NULL);  // needs to be cleared before flags changed
-    StringAttr _name;
-    src.read(_name);
+    setName(nullptr); // needs to be cleared before flags changed
+
+    src.read(deserializeContext.name);
     src.read(flags);
-    if (_name[0]==0) 
-        setName(NULL);
+    if (deserializeContext.name.isEmpty())
+        setName(nullptr);
```

Comment:
question: Is this needed I suspect it is already null.  Can it be replaced with dbgassertex(queryName()==nullptr);?